ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -3562,8 +3546,28 @@ unsigned HqlCppTranslator::buildRtlField(StringBuffer * instanceName, IHqlExpres
             break;
         }
 
-        if (xpathName.length() && (xpathName.charAt(0) == '@'))
-            typeFlags |= RFTMhasxmlattr;
+        if (xpathName.length())
+        {
+            if (xpathName.charAt(0) == '@')
+                typeFlags |= RFTMhasxmlattr;
+            if (!strpbrk(xpathName, "/?*[]<>"))
+                typeFlags |= RFTMxpathscalar;
+        }
+
+        StringBuffer typeName;
+        typeFlags = buildRtlType(typeName, fieldType, typeFlags); //benefit to adding other flags to generated code as well?
```

Comment:
Right I think most the flags have historically been checked as aggregates at the record level.  I saved off additonal xml related ones in case they might come in handy at the field level at some point but could limit it to the flag for scalar xpath for now if better.